plans/benchmarks/benchmarks.go

```diff
@@ -314,27 +338,74 @@ func SubtreeBench(runenv *runtime.RunEnv) error {
 
 		runenv.RecordMessage("i am a subscriber")
 
+		t := time.Now()
 		// if we are receiving, wait for the publisher to be done.
 		<-watcher.Barrier(ctx, handoff, int64(1))
 
+		metrics.GetOrRegisterResettingTimer("subscriber.after-barrier", nil).UpdateSince(t)
+
+		runenv.RecordMessage("after barrier")
+
+		var eg errgroup.Group
+
+		alltests := time.Now()
+
 		for _, tst := range tests {
-			ch := make(chan *string, 1)
-			err = watcher.Subscribe(ctx, tst.Subtree, ch)
-			if err != nil {
-				return err
-			}
-			for i := 1; i <= iterations; i++ {
-				t := prometheus.NewTimer(tst.Summary)
-				b := <-ch
-				t.ObserveDuration()
-				if strings.Compare(*b, *tst.Data) != 0 {
-					return fmt.Errorf("received unexpected value")
+			tst := tst
+			eg.Go(func() error {
+				randomSleepMs := rand.Int31n(10000)
+				time.Sleep(time.Duration(randomSleepMs) * time.Millisecond)
+
+				subscrtime := time.Now()
+				iserror := true
+
+				defer func() {
+					if iserror {
+						metrics.GetOrRegisterResettingTimer("subscriber.test.fail", nil).UpdateSince(subscrtime)
+					} else {
+						metrics.GetOrRegisterResettingTimer("subscriber.test.success", nil).UpdateSince(subscrtime)
+					}
+				}()
+
+				ch := make(chan *string, 1)
+				runenv.RecordMessage("subscribing (series: %s)", tst.Name)
+				err = watcher.Subscribe(ctx, tst.Subtree, ch)
+				if err != nil {
+					return err
 				}
-				if i%500 == 0 {
-					runenv.RecordMessage("received %d items (series: %s)", i, tst.Name)
+				runenv.RecordMessage("subscribed (series: %s)", tst.Name)
+				for i := 1; i <= iterations; i++ {
+					now := time.Now()
+					t := prometheus.NewTimer(tst.Summary)
+					b := <-ch
+					t.ObserveDuration()
+					if b == nil {
+						metrics.GetOrRegisterResettingTimer("subscriber.test.iteration.fail.nil", nil).UpdateSince(now)
+						return errors.New("nil b")
+					}
+
+					if strings.Compare(*b, *tst.Data) != 0 {
+						metrics.GetOrRegisterResettingTimer("subscriber.test.iteration.fail", nil).UpdateSince(now)
+						return fmt.Errorf("received unexpected value")
+					}
+
+					if i%500 == 0 {
+						runenv.RecordMessage("received %d items (series: %s)", i, tst.Name)
+					}
+					metrics.GetOrRegisterResettingTimer("subscriber.test.iteration.ok", nil).UpdateSince(now)
```

Comment:
@coryschwartz following my comment on your `influxdb metrics` PR - we have to think if we want to support that kind of measurements in the SDK. I personally find them useful, but I am fine if we don't.

However I am not sure it will be feasible for 1000 testplan instances, each running 10 tests (concurrently), going over 5000 iterations, to send individual measurements to InfluxDB without aggregating. This is just a single measurement, I imagine we would want to have 10-100 measurements per testplan.

In any case it is both experimenting with both approaches.